go/src/koding/kites/kloud/stack/provider/template.go

```diff
@@ -227,23 +228,37 @@ func (t *Template) FillVariables(prefix string) error {
 	return t.InjectVariables("", vars)
 }
 
-// InjectVariables
 func (t *Template) InjectVariables(prefix string, meta interface{}) error {
+	t.inject(prefix, meta)
+
+	return t.hclUpdate()
+}
+
+func (t *Template) InjectCredentials(creds ...*stack.Credential) error {
+	for _, cred := range creds {
+		t.inject(cred.Provider, cred.Credential)
+		t.inject(cred.Provider, cred.Bootstrap)
+	}
+
+	return t.hclUpdate()
+}
+
+func (t *Template) inject(prefix string, meta interface{}) {
+	t.log.Debug("injecting: %# v", meta)
+
 	for k, v := range t.b.New(prefix).Build(meta) {
 		// Ignore custom variables prefixed with __ from injecting.
 		// The ignored variables may contain interpolations which are
 		// not meant to be sent to Terraform.
 		if strings.HasPrefix(k, "custom___") {
-			t.log.Debug("Not injecting variable: %s=%s", k, v)
+			t.log.Debug("not injecting variable: %s=%s", k, v)
```

Comment:
we were trying to follow "errors are values/variables so they should start with small char, logs are sentences, should start with sentence case" rule.